### ISSUE TITLE
why use typeof(f(testvalue(data)))?

### ISSUE THREAD

**Issue by MasonProtter** [JuliaArrays/MappedArrays.jl#32]
Current, to determine the `eltype` of a `MappedArray`, this package computes `typeof(f(testvalue(data)))`. While it's commendable to try to avoid using `Core.Compiler.return_type` for this, I think it might be misguided. Not only could this be incorrect for heterogeneously typed arrays, but it's potentially very slow.

I'm very tempted to think this is a case where reaching into internals to just ask the compiler for the return type is justified.

**Comment by johnnychen94** [comment]
Just hit this issue... and I feel the result very weird

```julia
T = Union{Missing, Float32}
A = of_eltype(T, [missing, 3])
eltype(A) # missing

A = of_eltype(T, [3, missing])
eltype(A) # Float32
```

**Comment by timholy** [comment]
Yeah, I think inference is good enough to rely on now. @MasonProtter, want to submit a PR to fix this?

**Comment by johnnychen94** [comment]
closed by #43
